### PR TITLE
More support for STM32L4

### PR DESCRIPTION
**Pull request by davidgfnet**
I'm developing a firmware for STM32L496 and I find some stuff missing, so contributing back :D

### CODE REVIEW

**Comment by ChuckM** [comment]
This looks good although I'm worried about the naming scheme of `syscfg_common_<expanding-string>` that seems like it would become too unwieldly.

File: include/libopencm3/stm32/l4/exti.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/** @defgroup exti_defines EXTI Defines
+ *
+ * @brief <b>Defined Constants and Types for the STM32F4xx External Interrupts
+ * </b>
+ *
+ * @ingroup STM32F4xx_defines
```
Comment (markus-k) [inline include/libopencm3/stm32/l4/exti.h]: Shouldn't this say `STM32L4xx_defines`? Same a few lines above and in the syscfg header.

File: include/libopencm3/stm32/l4/exti.h [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * @author @htmlonly &copy; @endhtmlonly 2013
+ * Piotr Esden-Tempski <piotr@esden.net>
+ *
+ * @version 1.0.0
+ *
+ * @date 14 January 2014
```
Comment (karlp) [inline include/libopencm3/stm32/l4/exti.h]: None of the above is true.

File: include/libopencm3/stm32/l4/syscfg.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ * @ingroup STM32L1xx_defines
+ *
+ * @brief Defined Constants and Types for the STM32L1xx Sysconfig
+ *
+ * @version 1.0.0
+ *
+ * @author @htmlonly &copy; @endhtmlonly 2013
+ * Frantisek Burian <BuFran@seznam.cz>
+ *
+ * @date 13 January 2014
```
Comment (karlp) [inline include/libopencm3/stm32/l4/syscfg.h]: none of the above is true.

File: include/libopencm3/stm32/l4/syscfg.h [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * LGPL License Terms @ref lgpl_license
+ *  */
+
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2013 Frantisek Burian <BuFran@seznam.cz>
```
Comment (karlp) [inline include/libopencm3/stm32/l4/syscfg.h]: nope

**Comment by karlp** [comment]
exti looks good, and the rcc clock stuff, but the syscfg renaming is just a mess, don't bother with it.  it's pure chance that it wasn't already out of date.

File: include/libopencm3/stm32/l4/exti.h (hunk 1 above)
Comment (davidgfnet) [reply include/libopencm3/stm32/l4/exti.h]: Yeah, just copied :D Will fix

**Comment by davidgfnet** [comment]
So should I just leave the file name the way it is, or do we want it to be something like _common.h ?

**Comment by karlp** [comment]
Just leave the filenames at the moment.  It might be reasonable to rename it -v2, if it seems to apply to all the newer parts, but otherwise just leave it, those renames have been a maintennance nightmare.

**Comment by karlp** [comment]
You've still got copyrights and dates that are wrong.  Personally I don't consider "authors" of a one or two line include statement to be relevant at all, but I certainly feel it's misleading to have esden/bufran and 2013 listed in these files.

**Comment by davidgfnet** [comment]
So what should I change them for? I obviously copied the files, I won't add my name/copyright there. Tell me what you think they should have and I'll update the copyright notices.
Apart from that, anything else I'm missing to update in this pull request?

**Comment by br101** [comment]
Hi, sorry to jump in here, but I would like to get this (and more l4 support) merged.

Not exactly sure what's the policy in libopencm3 and what's acceptable for @karlp but as these files just include another file it's reasonable to simply remove the author (example stm32/sysconfig.h or l4/usart.h) - i think this is what @karlp implied in his comment. Otherwise it might be acceptable to just add your name and the current date.

Anything else that needs to be addressed to get this merged? Anything I can do to help?

**Comment by karlp** [comment]
IMO, you either put your own, or you drop them.  Attempting to say that the files were created in 2013 by someone else entirely is entirely bogus.  My vote would be to drop them, so no-one else tries this sort of thing in the future, but I'd accept your own name if you prefer.  Other options are a) wait until I get to just doing it myself, b) wait for someone else to make a PR that's ~same but different~ and c) wait for this one.

**Comment by davidgfnet** [comment]
Just removed the authors, can you go ahead and merge then?

**Comment by br101** [comment]
hi @davidgfnet did you push? i can not see the change in the pull request yet...

**Comment by davidgfnet** [comment]
I did push, what else do you need from me?

**Comment by br101** [comment]
sorry but syscfg is all wrong, exti incomplete, see my pull request https://github.com/libopencm3/libopencm3/pull/866

**Comment by br101** [comment]
I think this PR is obsolete by now

**Comment by karlp** [comment]
yep, looks like it's all in now.

**Pull request closed without merging**